apps/wallet-dashboard/components/AssetsList.tsx

```diff
@@ -16,12 +22,35 @@ const ASSET_LAYOUT: Record<AssetCategory, string> = {
     [AssetCategory.Other]: 'flex flex-col overflow-auto py-sm',
 };
 
-export function AssetList({ assets, selectedCategory }: AssetListProps): React.JSX.Element {
+export function AssetList({
+    assets,
+    selectedCategory,
+    hasNextPage = false,
+    isFetchingNextPage = false,
+    fetchNextPage,
+}: AssetListProps): React.JSX.Element {
+    const observerElem = useRef<HTMLDivElement | null>(null);
+    const { isIntersecting } = useOnScreen(observerElem);
+    const isSpinnerVisible = isFetchingNextPage && hasNextPage;
+
+    useEffect(() => {
+        if (isIntersecting && hasNextPage && !isFetchingNextPage && fetchNextPage) {
+            fetchNextPage();
+        }
+    }, [isIntersecting, fetchNextPage, hasNextPage, isFetchingNextPage]);
+
     return (
         <div className={ASSET_LAYOUT[selectedCategory]}>
             {assets.map((asset) => (
                 <AssetTileLink key={asset.digest} asset={asset} type={selectedCategory} />
             ))}
+            <div ref={observerElem}>
```

Comment:
wouldnt you use scroll for both screens? imo, we can include the scroll in the other assets category

![image](https://github.com/user-attachments/assets/b8d72740-3199-4b14-9d4c-2ec0497caf74)
![image](https://github.com/user-attachments/assets/2bff1634-4073-4067-8450-5623eaa3d6a0)